components/eventing-controller/pkg/handlers/jetstream.go

```diff
@@ -364,6 +318,98 @@ func getStreamConfig(natsConfig env.NatsConfig) (*nats.StreamConfig, error) {
 	return streamConfig, nil
 }
 
+func (js *JetStream) validateExistingSubscriptionFilters(subscription *eventingv1alpha1.Subscription, log *zap.SugaredLogger) error {
+	for key, jsSub := range js.subscriptions {
+		if !js.isJsSubAssociatedWithKymaSub(key, subscription) || !jsSub.IsValid() {
+			continue
+		}
+
+		// TODO: optimize this call of ConsumerInfo
+		// as jsSub.ConsumerInfo() will send an REST call to nats-server for each subject
+		info, err := jsSub.ConsumerInfo()
+		if err != nil {
+			if err == nats.ErrConsumerNotFound {
+				log.Infow("Deleting invalid Consumer!")
+				if err := js.deleteConsumerFromJetStream(key.ConsumerName(), log); err != nil {
+					return err
+				}
+				delete(js.subscriptions, key)
+				continue
+			}
+			return err
+		}
+
+		if !utils.ContainsString(js.GetJetStreamSubjects(subscription.Status.CleanEventTypes), info.Config.FilterSubject) {
+			if err := js.deleteSubscriptionFromJetStream(jsSub, key, log); err != nil {
+				return err
+			}
+			log.Infow(
+				"deleted JetStream subscription because it was deleted from subscription filters",
+				"subscriptionSubject", key,
+				"jetStreamSubject", jsSub.Subject,
+			)
+		}
+	}
+	return nil
+}
+
+func (js *JetStream) validateExistingSubscription(subscription *eventingv1alpha1.Subscription, asyncCallback func(m *nats.Msg), log *zap.SugaredLogger) {
```

Comment:
(nit) Does it make sense to rename this function to something which conveys that it binds consumers to JetStream subscription?